ADALiOS/ADALiOS/ADAuthenticationContext.m

```diff
@@ -813,7 +1089,7 @@ -(void) internalAcquireTokenWithResource: (NSString*) resource
         return;
     }
     
-    if (tryCache && promptBehavior != AD_PROMPT_ALWAYS && self.tokenCacheStore)
+    if (![self.class isForcedAuthorization:promptBehavior] && self.tokenCacheStore)
```

Comment:
self.class looks REALLY weird here, unless you want this to be overridable in a subclass I would use ADAuthenticationContext (or whatever class that's in...) instead.